### PR TITLE
Add `tz` support

### PR DESCRIPTION
**Pull request by pjrt**
Add a support for the `tz` library via `ltaoInTimeZoneTZ`

### CODE REVIEW

File: src/Canteven/ParseDate/Types.hs [hunk 1]
```diff
@@ -33,12 +36,20 @@ instance ParseTime LocalTimeAndOffset where
 
 -- | Get the localtime represented by this LTAO in the given timezone.
 ltaoInTimeZone :: TimeZoneSeries -> LocalTimeAndOffset -> LocalTime
+ltaoInTimeZone = ltaoInTimeZoneConvert utcToLocalTime'
+
+-- same as @ltaoInTimeZone@ but for TZ
```
Comment (glasserc) [inline src/Canteven/ParseDate/Types.hs]: We don't actually run Haddock, but you might use the Haddock style "doc comment" format: either `-- |` to document the following thing, or `-- ^` to document the preceding thing.

Comment (pjrt) [inline src/Canteven/ParseDate/Types.hs]: Opps, meant to add that.

File: src/Canteven/ParseDate/Types.hs [hunk 2]
```diff
@@ -33,12 +36,20 @@ instance ParseTime LocalTimeAndOffset where
...
+ltaoInTimeZoneConvert :: (a -> UTCTime -> LocalTime) -> a -> LocalTimeAndOffset -> LocalTime
 -- If the LTAO doesn't specify an offset, it must be local time, so just use it
 -- directly.
-ltaoInTimeZone _ LocalTimeAndOffset { ltaoLocalTime,
-                                      ltaoOffset = Nothing } = ltaoLocalTime
+ltaoInTimeZoneConvert _ _ (LocalTimeAndOffset lTime Nothing) = lTime
 -- Otherwise, this LTAO specifies a UTCTime, so convert that UTCTime to LocalTime.
-ltaoInTimeZone tz LocalTimeAndOffset { ltaoLocalTime, ltaoOffset = Just offset } =
-    utcToLocalTime' tz utc
+ltaoInTimeZoneConvert convert tz (LocalTimeAndOffset lTime (Just off)) =
```
Comment (glasserc) [inline src/Canteven/ParseDate/Types.hs]: I probably would have pulled the reformatting out into its own commit.

**Comment by glasserc** [comment]
Needs a version bump.

File: src/Canteven/ParseDate/Types.hs (hunk 2 above)
Comment (pjrt) [inline src/Canteven/ParseDate/Types.hs]: It looks like reformatting, but it is actually a rewriting. `ltaoInTimeZoneConvert` replaced the `ltaoInTimeZone` and github freaked out.
